modules/_labs/pagination/react/lib/Pagination/common/utils/safelySpreadProps.ts

```diff
@@ -0,0 +1,20 @@
+import {flex} from './flex';
+import {layout} from './layout';
+import {space} from './space';
+
+const styleProps = {
+  ...flex,
+  ...layout,
+  ...space,
+};
+
+// TODO: Improve this function
+export function safelySpreadProps<T extends object>(props: T) {
+  let safeProps = {};
+  for (const key in props) {
+    if (!(key in styleProps)) {
+      safeProps = {...safeProps, [key]: props[key as keyof T]};
```

Comment:
Mutation is safe within a function's scope if the function owns the variable. It will produce less garbage as well:

```suggestion
      safeProps[key] = props[key as keyof T]};
```

This way the `let safeProps` can be a `const safeProps`

Of course there's tests to make sure such a refactor won't break the specification of this function, right?

Reply:
That's a good call out instead of creating a new obj on every iteration. I updated it 👍🏻